components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -519,7 +528,9 @@ void AdsServiceImpl::ShutdownAndResetState() {
   VLOG(6) << "Resetting ads state";
 
   profile_->GetPrefs()->ClearPrefsWithPrefixSilently("brave.brave_ads");
-  local_state_->ClearPref(brave_l10n::prefs::kCountryCode);
+  if (local_state_) {
```

Comment:
this gets called during unit tests?

Reply:
I do not recall tbh if this was crashing or I just added this pre-emptively. I can remove and retest.

Reply:
After testing, no `AdsServiceImpl::ShutdownAndResetState` doesn't get called in any of the unit tests. I can remove the `if` here.

Reply:
Removed in [1f0d7ea](https://github.com/brave/brave-core/pull/20433/commits/1f0d7ea6649c3bd7cd9278b274828df216cae4c1)